tests/test.js

```diff
@@ -0,0 +1,108 @@
+/* global it */
+
+'use strict';
+
+var assert = require('assert');
+var path = require('path');
+var gulp = require('gulp');
+
+var notes = require('..');
+
+it('Should be a function', function () {
+  assert.strictEqual(typeof notes, 'function');
+});
+
+it('Should create the `notes.md` file', function (done) {
+  var files = [];
+
+  gulp.src([
+    path.join(__dirname, 'fixture', 'fixture.js'),
+    path.join(__dirname, 'fixture', 'fixture2.js')
+  ])
+
+  .pipe(notes())
+
+  .on('data', function (data) {
+    files.push(data);
+  })
+
+  .on('end', function () {
+    assert.strictEqual(files.length, 1);
+    assert.strictEqual(files[0].relative, 'notes.md');
+
+    done();
+  });
+});
+
+it('`notes.md` file should get generated correctly', function (done) {
+  var files = [];
+
+  gulp.src([
+    path.join(__dirname, 'fixture', 'fixture.js'),
+    path.join(__dirname, 'fixture', 'fixture2.js')
+  ])
+  .pipe(notes())
+  .on('data', function (data) {
+    files.push(data);
+  })
+
+  .on('end', function () {
+    var contents = files[0].contents.toString();
+
+    assert.strictEqual(contents.split('\n')[0], '# Notes', 'Should have a header');
+    assert.strictEqual(contents.split('\n')[2], '## TODO');
+    assert.strictEqual(contents.split('\n')[3], '* Write more tests - **fixture.js:1**');
+    assert.strictEqual(contents.split('\n')[8], '## ARTHUR');
+    assert.strictEqual(contents.split('\n')[9], '* Do this - **fixture.js:3**');
```

Comment:
Assign the `contents.split('\n')` to a variable. easier to read and less typing.

Alternatively this is a little brittle because the output is dependent on the ordering. Perhaps an `indexOf(...) !== -1` test would be better.